kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,70 @@
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            'A similar string to "%s" exists: "%s"',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    def __init__(self, linter):
+        super().__init__(linter)
+        self._dict_of_strings = {}
+        self.threshold = 0.8
+
+    @staticmethod
+    def clean_string(text):
+        for word in text:
+            if word not in string.punctuation:
+                text += "".join(word)
+        return text
+
+    @staticmethod
+    def marked_for_translation(funcname):
+        return funcname in ("_", "gettext_lazy")
+
+    def check_similar_string(self, translation_string):
+        for key in self._dict_of_strings:
+            if (
+                levenshtein.normalized_similarity(
+                    self.clean_string(translation_string), self.clean_string(key)
+                )
+                >= self.threshold
+            ):
+                return key
+        return None
+
+    def check_same_string(self, translation_string):
+        return translation_string in self._dict_of_strings
+
+    @utils.check_messages("similar-string")
+    def visit_call(self, node):
+        if self.marked_for_translation(node.func.as_string()) and isinstance(
+            node.args[0], astroid.nodes.Const
+        ):
```

Comment:
Instead of nested if conditions I prefer to have an if-return sequence here. It makes the algorithm more easier to follow in case it needs to be changed later.

Reply:
```
if not self.marked_for_translation():
    return

if not ...
    return


```